components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -614,32 +632,36 @@ private void addActionPropertiesToDB(String actionId, Map<String, String> action
      * @return A map of action properties, including any additional data based on action type.
      * @throws ActionMgtException If an error occurs while retrieving action properties from the database.
      */
-    private Map<String, String> getActionPropertiesFromDB(String actionId, Integer tenantId) throws ActionMgtException {
+    private Map<String, ActionPropertyForData> getActionPropertiesFromDB(String actionId, Integer tenantId)
+            throws ActionMgtException {
 
         NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
-        Map<String, String> actionEndpointProperties = new HashMap<>();
-
+        Map<String, ActionPropertyForData> actionEndpointProperties = new HashMap<>();
         try {
-            boolean primitiveValueColumnExists = isPrimitiveValueColumnExists();
-            String query = primitiveValueColumnExists ? ActionMgtSQLConstants.Query.GET_ACTION_PROPERTIES_INFO_BY_ID
-                    : ActionMgtSQLConstants.Query.GET_ACTION_PROPERTIES_INFO_BY_ID_WITH_PROPERTY_VALUE_COLUMN;
+            String query = ActionMgtSQLConstants.Query.GET_ACTION_PROPERTIES_INFO_BY_ID;
             jdbcTemplate.withTransaction(template ->
                 template.executeQuery(query,
                     (resultSet, rowNumber) -> {
-                        actionEndpointProperties.put(
-                                resultSet.getString(ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PROPERTY_NAME),
-                                resultSet.getString(primitiveValueColumnExists ?
-                                        ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PRIMITIVE_VALUE :
-                                        ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PROPERTY_VALUE));
+                        if (resultSet.getString(ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PROPERTY_TYPE)
+                                .equals(ActionProperty.Type.PRIMITIVE.toString())) {
```

Comment:
```suggestion
                        if (ActionProperty.Type.PRIMITIVE.name()                 .equals(resultSet.getString(ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PROPERTY_TYPE))) {
```

Let's change like this. This method is susceptible for NPE

Reply:
Updated in